zne/extrapolation/extrapolator.py

```diff
@@ -15,96 +15,123 @@
+        if num_points < self.min_points:
+            raise ValueError(
+                f"Insufficient number of distinct data points provided ({num_points}), "
+                f"at least {self.min_points} needed."
+            )
+        sizes = (x_data.size, y_data.size, sigma_x.size, sigma_y.size)
+        if any(s != x_data.size for s in sizes):
+            raise ValueError(
+                "Invalid data, all inputs should be of the same size: "
+                f"{x_data.size = }, {y_data.size = }, {sigma_x.size = }, {sigma_y.size = }."
+            )
+        return self._extrapolate_zero(x_data, y_data, sigma_x, sigma_y)
 
     ################################################################################
     ## VALIDATION
     ################################################################################
-    def _validate_data(self, data: Sequence[Sequence[float]]) -> tuple[RegressionDatum, ...]:
-        """Validates data points for the regression model.
+    def _validate_data(
+        self, data: Sequence[float] | "ndarray[Any, dtype[float_]]"
+    ) -> "ndarray[Any, dtype[float_]]":
+        """Validates data for the regression model.
 
         Args:
-            data: A sequence of data points to fit the regression model from, each of which
-                consisting on the value of the regressor, measured mean, and estimated variance.
+            data: A sequence of values for the regression model.
 
         Returns:
             Validated data normalized.
-
-        Raises:
-            TypeError: If provided data is not a valid sequence of data points.
-            ValueError: If the number of data points is smaller than the minimum required
-                number of data points for the given Estrapolator.
         """
-        if not isinstance(data, Sequence) or not all(
-            isinstance(d, Sequence) and len(d) == 3 and all(isinstance(v, (float, int)) for v in d)
-            for d in data
-        ):
-            raise TypeError(
-                "Invalid data provided, expeceted Sequence[Sequence[float, float, float]]."
-            )
-        if len(data) < self.min_points:
+        try:
+            data = array(data).astype(float_, casting="same_kind")
+        except TypeError as exc:
+            raise TypeError(f"Invalid non-numeric (i.e. non-real) data array: {data}.") from exc
+        if len(data.shape) != 1:
             raise ValueError(
-                f"At least {self.min_points} data point{'s' if self.min_points != 1 else ''} "
-                "must be provided to fit the regression model."
+                "Invalid regression data provided, expected one dimensional sequence of floats."
             )
-        return tuple(tuple(d) for d in data)  # type: ignore
+        return data
+
+    def _validate_sigma(
+        self,
+        sigma: Sequence[float] | "ndarray[Any, dtype[float_]]" | None,
+        default_size: int,
+    ) -> "ndarray[Any, dtype[float_]]" | None:
```

Comment:
Have you looked at the `NPTyping` module? Mildly annoying to add a dependency, but it's lightweight and makes numpy type hints much nicer imo
[
](https://pypi.org/project/nptyping/)

Reply:
Newer versions of python don't need it so I was annoyed when everything started breaking... let me explore it though, great suggestion!

Reply:
I have moved away from numpy a bit more, so for the two functions that still use it, I have decided to not add an extra dependency, this is a good thing to keep in mind for the future though! 

Thanks a lot 🙂